content/blog/an-example-application-for-processing-stock-market-trade-data-on-the-mapr-converged-data-platform.md

```diff
@@ -0,0 +1,291 @@
+
+There are several beneficial aspects of the application that are worth highlighting:
+
+* The Consumer microservice, performing the indexing, can be arbitrarily scaled simply by running more instances.  See below in this README for how to start the application.
+* Jackson annotations are provided for easy translation of the data structures to JSON and persistence to MapR-DB.
+* The application can handle 300,000 entries/second on a 3-node cluster, which is suitable for testing.  It does not require a large cluster and takes advantage of the scaling properties of MapR Streams.
+* The resulting index topics are small and can be queried fast enough such that they can be used for interactive dashboards, such as in a Zeppelin notebook.
+
+## Architecture
+
+The application provides a financial intermediary service, running *bids* and *asks* between traders.  Traders are identified with a unique ID and each bid and ask is sent from one trader to a set of N receiving traders.
+
+The following diagram shows how data moves through the architecture. The rounded rectangles represent processes that produce and/or consume data from MapR Streams topics. Java-based microservices are used to ingest and manipulate streaming data using the Kafka API. Spark and Apache Zeppelin are used to provide streaming analytics and batch-oriented visualization.
+
+<img src="https://github.com/mapr-demos/finserv-application-blueprint/raw/master/images/dataflow.gif" width="70%">
+
+## Prerequisites
+
+This application requires MapR 5.2 and Spark 2.0, which can be easily installed with the [MapR Ecosystem Pack 2.0](http://maprdocs.mapr.com/home/InteropMatrix/r_MEP_52.html).
+You can use MapR's free [Converged Community Edition](http://mapr.com/download) or the [Converged Enterprise Edition](https://www.mapr.com/products/mapr-distribution-editions).   
+
+You will also need Git and Apache Maven in order to download and compile the provided source code.
+
+## Building the application
+
+Clone this repo and build the application with Maven.  A pom.xml file is included in the base directory. The remainder of this guide will assume that you clone the package to /home/mapr/.
+
+```
+cd /home/mapr/
+git clone http://github.com/mapr-demos/finserv-application-blueprint.git
+cd finserv-application-blueprint
+mvn clean install
+```
+
+At this point you should see the resulting jar file in the target/ directory:  `nyse-taq-streaming-1.0.jar`
+
+Copy that jar package to the /home/mapr/ directory on each of your cluster nodes:
+
+```
+scp ./target/nyse-taq-streaming-1.0.jar mapr@<YOUR_MAPR_CLUSTER>:/home/mapr
+```
+
+### Step 1: Create the stream
+
+A *stream* is a logical grouping of topics. They give us a way to group together topics and protect those topics with a single set of security permissions and other properties. MapR supports the Kafka API for interacting with streams.  For more information on Streams, see <https://www.mapr.com/products/mapr-streams>.
```

Comment:
Note the URL www.mapr.com/products/mapr-streams redirects readers to https://www.hpe.com/us/en/software/ezmeral-data-fabric.html. Is it what you want?

In this section (or maybe in the introductory paragraph) you may also want to make a reference to the post here: https://developer.hpe.com/blog/real-time-event-streaming-what-are-your-options/